File: back-end/aths/src/config/email.js
import nodemailer from 'nodemailer';
import dotenv from 'dotenv';

dotenv.config();

/**
 * Email Configuration
 * Configures Nodemailer for sending emails via Gmail
 */

/**
 * Create email transporter
 */
const createTransporter = () => {
  return nodemailer.createTransport({
    service: process.env.EMAIL_SERVICE || 'gmail',
    host: process.env.EMAIL_HOST || 'smtp.gmail.com',
    port: parseInt(process.env.EMAIL_PORT) || 587,
    secure: process.env.EMAIL_SECURE === 'true',
    auth: {
      user: process.env.EMAIL_USER,
      pass: process.env.EMAIL_PASSWORD,
    },
  });
};

/**
 * Get email from address
 */
export const getFromAddress = () => {
  const fromName = process.env.EMAIL_FROM_NAME || 'R-MAN E-Commerce';
  const fromEmail = process.env.EMAIL_FROM || process.env.EMAIL_USER;
  return `"${fromName}" <${fromEmail}>`;
};

/**
 * Verify email configuration
 */
export const verifyEmailConfig = async () => {
  try {
    const transporter = createTransporter();
    await transporter.verify();
    console.log('✅ Email service is ready');
    return true;
  } catch (error) {
    console.error('❌ Email service verification failed:', error.message);
    return false;
  }
};

export const transporter = createTransporter();

export default {
  transporter,
  getFromAddress,
  verifyEmailConfig,
};

File: back-end/cmps/app/config/email.py
"""
Email Configuration
Configures SMTP for sending emails
"""
import smtplib
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart
from app.config.settings import settings
import logging

logger = logging.getLogger(__name__)


def send_email(to_email: str, subject: str, html_body: str, text_body: str = None):
    """
    Send email via SMTP
    
    Args:
        to_email: Recipient email address
        subject: Email subject
        html_body: HTML email body
        text_body: Plain text email body (optional)
    
    Returns:
        bool: True if email sent successfully, False otherwise
    """
    try:
        # Create message
        msg = MIMEMultipart('alternative')
        msg['From'] = f"{settings.EMAIL_FROM_NAME} <{settings.EMAIL_FROM}>"
        msg['To'] = to_email
        msg['Subject'] = subject
        
        # Add text part if provided
        if text_body:
            msg.attach(MIMEText(text_body, 'plain'))
        
        # Add HTML part
        msg.attach(MIMEText(html_body, 'html'))
        
        # Connect to SMTP server
        with smtplib.SMTP(settings.EMAIL_HOST, settings.EMAIL_PORT) as server:
            server.starttls()
            server.login(settings.EMAIL_USER, settings.EMAIL_PASSWORD)
            server.send_message(msg)
        
        logger.info(f"📧 Email sent successfully to {to_email}: {subject}")
        return True
        
    except Exception as e:
        logger.error(f"❌ Failed to send email to {to_email}: {str(e)}")
        return False


def verify_email_config():
    """
    Verify email configuration by testing SMTP connection
    
    Returns:
        bool: True if configuration is valid, False otherwise
    """
    try:
        with smtplib.SMTP(settings.EMAIL_HOST, settings.EMAIL_PORT, timeout=5) as server:
            server.starttls()
            server.login(settings.EMAIL_USER, settings.EMAIL_PASSWORD)
        
        logger.info("✅ Email service configuration verified")
        return True
        
    except Exception as e:
        logger.warning(f"⚠️  Email service verification failed: {str(e)}")
        logger.warning("Email notifications will be disabled")
        return False
